### ISSUE TITLE
Frequent crashes when running over forwarded X or when running in Docker.

### ISSUE THREAD

**Issue by AaronFriel** [atom/atom#8684]
When using a variety of Windows (as in Vista, 7, 10) X Servers (e.g.: the server packaged with MobaXTerm), Atom crashes very frequently and logs this:

```
Failed to get crash dump id.
Report Id:
```

Following this I'm prompted if I would like to close the window or reload, but either way I lose the session state I had. (Any unsaved changes are lost, what windows I've opened are lost, any directories added to the project are lost.)

This is obviously not a very helpful error message to log, and the "Reload" option is very unsatisfactory in that it fails to recover any information. The crashes are very frequent and occur as a result of a wide variety of things, including using hotkeys (ctrl-P, ctrl-shift-P, ctrl-shift-I), hovering over text that might invoke a plugin, et cetera. Disabling plugins doesn't seem to help, and Atom crashes frequently in this setting.

I am using Atom in Docker, using Ubuntu 14.04 as a base image, with all packages up to date, and Atom's dependencies up to date. (Is there a list of optional Atom dependencies somewhere? e.g.: Atom reports an error connecting to PulseAudio every time it starts.)

Periodically I get this error as well in the log, and Atom crashes immediately with:

```
Failed to generate minidump.Failed to get crash dump id.
Report Id:
```

In this case, I get no prompts and Atom closes immediately.

The only packages I have enabled are (from `apm list`):

```
/home/developer/.atom/packages (5)
├── adventurous-syntax@0.2.0
├── autocomplete-haskell@0.4.5
├── haskell-ghc-mod@0.8.14
├── ide-haskell@0.7.1
└── language-haskell@1.4.1
```

**Comment by AaronFriel** [comment]
I wanted to check if the crash was due to the Ubuntu 14.04 image I used or the older kernel on the Docker host (Ubuntu 15.04).

Today I duplicated the crash on Arch Linux (uname: `Linux reactr-dev 4.1.6-1-ARCH #1 SMP PREEMPT Mon Aug 17 08:52:28 CEST 2015 x86_64 GNU/Linux`) using the latest package for Atom in AUR. That would be 1.0.11-1 from [AUR](https://aur.archlinux.org/packages/atom-editor/).

**Comment by AaronFriel** [comment]
@50Wliu 

Here is the result of me opening up a folder in `--safe` mode, and opening a file and doing "Ctrl-Shift-I" repeatedly until I get this message:

"The editor has crashed
Please report this issue to https://github.com/atom/atom"

```
$ atom --safe -f -l ./eff/
[35975:0908/043736:ERROR:browser_main_loop.cc(170)] Running without the SUID sandbox! See https://code.google.com/p/chromium/wiki/LinuxSUIDSandboxDevelopment for more information on developing with the sandbox on.
App load time: 757ms
[35975:0908/043738:INFO:CONSOLE(175)] "Enabled theme 'adventurous-syntax' is not installed.", source: /usr/share/atom/resources/app.asar/src/theme-manager.js (175)
[35975:0908/043739:INFO:CONSOLE(175)] "Enabled theme 'adventurous-syntax' is not installed.", source: /usr/share/atom/resources/app.asar/src/theme-manager.js (175)
[35975:0908/043739:INFO:CONSOLE(175)] "Enabled theme 'adventurous-syntax' is not installed.", source: /usr/share/atom/resources/app.asar/src/theme-manager.js (175)
[35975:0908/043739:INFO:CONSOLE(46)] "Window load time: 1696ms", source: file:///usr/share/atom/resources/app.asar/static/index.js (46)
[35975:0908/043744:ERROR:channel.cc(305)] RawChannel read error (connection broken)
[35975:0908/043750:ERROR:channel.cc(305)] RawChannel read error (connection broken)
[35975:0908/043751:ERROR:channel.cc(305)] RawChannel read error (connection broken)
[35975:0908/043752:ERROR:CONSOLE(1)] "Uncaught ReferenceError: DevToolsAPI is not defined", source:  (1)
[35975:0908/043752:ERROR:channel.cc(305)] RawChannel read error (connection broken)
[35975:0908/043753:ERROR:channel.cc(305)] RawChannel read error (connection broken)
```

I am doing X Forwarding over IP (no SSH) from a VM to a Windows host. Crashes and hangs occur regardless of X server configuration (either hardware accelerated OpenGL, software OpenGL, or no OpenGL.)

Here is the result of doing the same set of actions after disabling OpenGL on the X server:

Verifying OpenGL is disabled on the display:
```
$ glxinfo
name of display: 10.0.0.1:0.0
Error: couldn't find RGB GLX visual or fbconfig
```

```
$ atom --safe -f -l ./eff/
[36112:0908/044244:ERROR:browser_main_loop.cc(170)] Running without the S
UID sandbox! See https://code.google.com/p/chromium/wiki/LinuxSUIDSandbox
Development for more information on developing with the sandbox on.
App load time: 165ms
[36112:0908/044245:INFO:CONSOLE(175)] "Enabled theme 'adventurous-syntax'
 is not installed.", source: /usr/share/atom/resources/app.asar/src/theme
-manager.js (175)
[36112:0908/044246:INFO:CONSOLE(175)] "Enabled theme 'adventurous-syntax'
 is not installed.", source: /usr/share/atom/resources/app.asar/src/theme
-manager.js (175)
[36112:0908/044246:INFO:CONSOLE(175)] "Enabled theme 'adventurous-syntax'
 is not installed.", source: /usr/share/atom/resources/app.asar/src/theme
-manager.js (175)
[36112:0908/044246:INFO:CONSOLE(46)] "Window load time: 1678ms", source:
file:///usr/share/atom/resources/app.asar/static/index.js (46)
[36112:0908/044254:ERROR:channel.cc(305)] RawChannel read error (connecti on broken)
[36112:0908/044254:ERROR:channel.cc(305)] RawChannel read error (connecti
on broken)
```
This repeats many times as `Ctrl-Shift-I` appears to be a nondeterministic path to reproduce. What finally caused the crash was me doing `Ctrl-,`:

```
[36112:0908/044326:ERROR:channel.cc(305)] RawChannel read error (connecti
on broken)
[36112:0908/044327:ERROR:channel.cc(305)] RawChannel read error (connecti
on broken)
[36112:0908/044327:ERROR:channel.cc(305)] RawChannel read error (connecti
on broken)
Crash dump id: ebf3f4b0-6bbd-4271-ac3e-a380cb06d286
/usr/bin/atom: line 99: 36112 Bus error               (core dumped) "$ATO
M_PATH" --executed-from="$(pwd)" --pid=$$ "$@"
```

In this instance, I actually had the editor crash instead of hang, even without OpenGL.

The crash is very reliable, repeatedly doing `Ctrl-Shift-I` seems to trigger it within the first ten or so attempts:

```
$ atom --safe -f -l ./eff/
[36614:0908/044809:ERROR:browser_main_loop.cc(170)] Running without the SUID sandbox! See https://code.google.com/p/chromium/wiki/LinuxSUIDSandboxDevelopment for more information on developing with the sandbox on.
App load time: 490ms
[36614:0908/044810:INFO:CONSOLE(175)] "Enabled theme 'adventurous-syntax' is not installed.", source: /usr/share/atom/resources/app.asar/src/theme-manager.js (175)
[36614:0908/044811:INFO:CONSOLE(175)] "Enabled theme 'adventurous-syntax' is not installed.", source: /usr/share/atom/resources/app.asar/src/theme-manager.js (175)
[36614:0908/044811:INFO:CONSOLE(175)] "Enabled theme 'adventurous-syntax' is not installed.", source: /usr/share/atom/resources/app.asar/src/theme-manager.js (175)
[36614:0908/044811:INFO:CONSOLE(46)] "Window load time: 1557ms", source: file:///usr/share/atom/resources/app.asar/static/index.js (46)
[36614:0908/044821:ERROR:channel.cc(305)] RawChannel read error (connection broken)
[36614:0908/044822:ERROR:channel.cc(305)] RawChannel read error (connection broken)
[36614:0908/044822:ERROR:channel.cc(305)] RawChannel read error (connection broken)
[36614:0908/044823:ERROR:channel.cc(305)] RawChannel read error (connection broken)
[36614:0908/044825:ERROR:channel.cc(305)] RawChannel read error (connection broken)
[36614:0908/044825:ERROR:channel.cc(305)] RawChannel read error (connection broken)
[36614:0908/044827:ERROR:channel.cc(305)] RawChannel read error (connection broken)
[36614:0908/044828:ERROR:channel.cc(305)] RawChannel read error (connection broken)
[36614:0908/044828:ERROR:channel.cc(305)] RawChannel read error (connection broken)
[36614:0908/044829:ERROR:channel.cc(305)] RawChannel read error (connection broken)
[36614:0908/044830:ERROR:channel.cc(305)] RawChannel read error (connection broken)
[36614:0908/044831:ERROR:channel.cc(305)] RawChannel read error (connection broken)
[36614:0908/044832:ERROR:channel.cc(305)] RawChannel read error (connection broken)
[36614:0908/044833:ERROR:channel.cc(305)] RawChannel read error (connection broken)
[36614:0908/044834:ERROR:channel.cc(305)] RawChannel read error (connection broken)
Crash dump id: 46dfa4b1-3fa8-4131-95a7-16f405ab2ce4
/usr/bin/atom: line 99: 36614 Bus error               (core dumped) "$ATOM_PATH" --executed-from="$(pwd)" --pid=$$ "$@"
```

**Comment by AaronFriel** [comment]
Is there anything else I can do to help the analysis?

**Comment by jrgifford** [comment]
Anything I can do to help with this?

**Comment by benogle** [comment]
cc @zcbenz

**Comment by AaronFriel** [comment]
@jrgifford, @benogle: Thanks for replying to this thread. I would love to help you help on this, and I can get crash dumps, and I can replicate on Ubuntu, Fedora, and Arch. I am not *intimately* familiar with obtaining crash dumps on each platform, or how you'd like to go about reproducing. I could provide more information here or over email.

**Comment by jrgifford** [comment]
@AaronFriel OK! Can you send us replication details here, and then I can dive into it.

**Comment by ulrichSchreiner** [comment]
hi,

i bundled atom and a few plugin's to have [atocker](https://github.com/ulrichSchreiner/atocker), a nice little Go environment. Everythin runs in a docker container. 

Working with this setup is complicated because you regularly get a "Failed to generate minidump" and atom crashes. When working on Go projects it happens at least once an hour. It would be great to know what to do do have a higher stability.

thx

**Comment by mborejdo** [comment]
I am experiencing the same crashes with Atom (1.4.0) inside Docker (1.9.1) on Ubuntu 15.10 with 4.2.0-23 Kernel.

**Comment by thenewvu** [comment]
Hello guys,

I've experienced this issue frequently in my docker `ubuntu:wily` running under a non-root user, the only thing I got so far is:

```
Failed to generate minidump. Failed to get crash dump id.
```

But after mounted `/dev/shm` (by adding `-v /dev/shm:/dev/shm`) the crash hasn't been  issued so far. I don't write that this is a solution for the issue, just a possible clue that we can follow. Hope you guys give some time to test this clue with me, for further investigation, I think we need the core team get involved.

**Comment by ulrichSchreiner** [comment]
hi,

i didn't have this issue since a few months. and i do not mount `/dev/shm`. my startup script for my container is [here](https://github.com/ulrichSchreiner/atocker/blob/master/atocker)

**Comment by thenewvu** [comment]
Thank for your answer, @ulrichSchreiner. Seems the issue comes from somewhere else.

**Comment by jomeier** [comment]
For me @thenewvu 's answer works also!

**Comment by endophage** [comment]
Confirming mounting `/dev/shm` fixed problem for me on Ubuntu 16.04 with Docker 1.11.

Incidentally this seems like an Electron problem, not Atom specifically as I've been having precisely the same issue with slack and it fixed it there too.

**Comment by lucasvanlierop** [comment]
I experience the same issues when running Slack 2.1.1. on debian:stretch in Docker. I'm running with `--ipc host` now and it hasn't crashed so far. See: https://docs.docker.com/engine/reference/run/#/ipc-settings-ipc
